datastore/tests/unit/test_entity.py

```diff
@@ -192,6 +193,13 @@ def test__eq__same_value_different_meanings(self):
 
         self.assertFalse(entity1 == entity2)
 
+    def test_id(self):
```

Comment:
You also need a test for the case when it is `None`

Reply:
It was in the constructor test.  I can add one I guess.

Reply:
Please do.